integration_tests/cellarfees_test.go

```diff
@@ -0,0 +1,418 @@
+
+	chainID, err := ethClient.NetworkID(context.Background())
+	if err != nil {
+		return err
+	}
+
+	signedTx, err := ethereumtypes.SignTx(tx, ethereumtypes.NewEIP155Signer(chainID), privateKey)
+	if err != nil {
+		return err
+	}
+
+	err = ethClient.SendTransaction(context.Background(), signedTx)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *IntegrationTestSuite) TestCellarFees() {
+	s.Run("Bring up chain, send fees from ethereum, observe auction and fee distribution", func() {
+		val := s.chain.validators[0]
+		ethereumSender := val.ethereumKey.address
+		kb, err := val.keyring()
+		s.Require().NoError(err)
+
+		ethClient, err := ethclient.Dial(fmt.Sprintf("http://%s", s.ethResource.GetHostPort("8545/tcp")))
+		s.Require().NoError(err)
+
+		clientCtx, err := s.chain.clientContext("tcp://localhost:26657", &kb, "val", val.keyInfo.GetAddress())
+		s.Require().NoError(err)
+
+		auctionQueryClient := auctiontypes.NewQueryClient(clientCtx)
+		bankQueryClient := banktypes.NewQueryClient(clientCtx)
+		cellarfeesQueryClient := cellarfeestypes.NewQueryClient(clientCtx)
+		corkQueryClient := corktypes.NewQueryClient(clientCtx)
+		distQueryClient := disttypes.NewQueryClient(clientCtx)
+
+		s.T().Log("Verify that the first validator address is an approved cellar ID")
+		idsRes, err := corkQueryClient.QueryCellarIDs(context.Background(), &corktypes.QueryCellarIDsRequest{})
+		s.Require().NoError(err)
+
+		var found bool
+		for _, id := range idsRes.CellarIds {
+			if id == ethereumSender {
+				found = true
+				break
+			}
+		}
+		s.Require().True(found, "validator ethereum address %s is not an approved cellar ID", ethereumSender)
+
+		s.T().Logf("Verify that the module account's fee balances are zero")
+		ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+		defer cancel()
+
+		acctsRes, err := cellarfeesQueryClient.ModuleAccounts(ctx, &cellarfeestypes.QueryModuleAccountsRequest{})
+		s.Require().NoError(err, "Failed to query module accounts")
+
+		feesAddress := acctsRes.FeesAddress
+		s.T().Logf("Fees address: %s", feesAddress)
+		balanceRes, err := bankQueryClient.Balance(ctx, &banktypes.QueryBalanceRequest{
+			Address: feesAddress,
+			Denom:   fmt.Sprintf("gravity%s", alphaERC20Contract.Hex()),
+		})
+		s.Require().NoError(err, "Failed to query fee balance of denom %s", alphaERC20Contract.Hex())
+		s.Require().Zero(balanceRes.Balance.Amount.Uint64())
+		balanceRes, err = bankQueryClient.Balance(ctx, &banktypes.QueryBalanceRequest{
+			Address: feesAddress,
+			Denom:   fmt.Sprintf("gravity%s", betaERC20Contract.Hex()),
+		})
+		s.Require().NoError(err, "Failed to query fee balance of denom %s", betaERC20Contract.Hex())
+		s.Require().Zero(balanceRes.Balance.Amount.Uint64())
+
+		s.T().Log("Getting baseline rewards increase rate")
+		rewardsRes, err := distQueryClient.DelegationRewards(ctx, &disttypes.QueryDelegationRewardsRequest{
```

Comment:
Is this inflation in the test environment mucking with outputs? The prod chain has inflation disabled, we can also disable it in the test environment to avoid having to be concerned with emissions that won't be happening.